src/sql-parser/src/parser.rs

```diff
@@ -2200,13 +2200,39 @@ impl<'a> Parser<'a> {
         Ok(CsrConnectionProtobuf { connection, seed })
     }
 
+    fn parse_source_error_policy_option(&mut self) -> Result<SourceErrorPolicy, ParserError> {
+        match self.expect_one_of_keywords(&[INLINE])? {
+            INLINE => Ok(SourceErrorPolicy::Inline),
+            _ => unreachable!(),
+        }
+    }
+
     fn parse_source_envelope(&mut self) -> Result<SourceEnvelope, ParserError> {
         let envelope = if self.parse_keyword(NONE) {
             SourceEnvelope::None
         } else if self.parse_keyword(DEBEZIUM) {
             SourceEnvelope::Debezium
         } else if self.parse_keyword(UPSERT) {
-            SourceEnvelope::Upsert
+            let value_decode_err_policy = if self.consume_token(&Token::LParen) {
+                // We only support the `VALUE DECODING ERRORS` option for now, but if we add another
+                // we should extract this into a helper function.
+                self.expect_keywords(&[VALUE, DECODING, ERRORS])?;
+                let _ = self.consume_token(&Token::Eq);
+                let open_inner = self.consume_token(&Token::LParen);
```

Comment:
I feel like for consistency with the other options we should ditch the parens for now?

```
VALUE DECODING ERRORS = INLINE
```

If we do need to ultimately support `(INLINE, PROPAGATE)`, it's very easy to support both syntaxes:

```
VALUE DECODING ERRORS = INLINE
VALUE DECODING ERRORS = (INLINE)
VALUE DECODING ERRORS = (INLINE, PROPAGATE)
```

You just promote the bare `INLINE` to `(INLINE)`.

Reply:
this logic already supports both
```
VALUE DECODING ERRORS = INLINE
VALUE DECODING ERRORS = (INLINE)
```
since this line you linked optionally consumes the inner parenthesis. See the sql-parser test additions

Reply:
Sorry, one of those days. 🤦🏽‍♂️ Carry on!